go/logic/command_applier.go

```diff
@@ -45,6 +45,8 @@ func (applier *CommandApplier) ApplyCommand(op string, value []byte) interface{}
 		return applier.registerNode(value)
 	case "discover":
 		return applier.discover(value)
+	case "async-discover":
+		return applier.asyncDiscover(value)
```

Comment:
@shlomi-noach should publishDiscoverMasters also use async-discover? 
because sync discover can slow down the raft log apply。
cc @yangeagle